server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java

```diff
@@ -202,13 +210,14 @@ public Invocations getInvocations() {
          cachesHeath.put(ch.getCacheName(), ch.getStatus());
       }
 
+      RestCacheManager<Object> restCacheManager = invocationHelper.getRestCacheManager();
       // We rely on the fact that getCacheNames doesn't block for embedded - remote it does unfortunately
-      return Flowable.fromIterable(cacheManager.getCacheNames())
-            .map(cacheManager::getCache)
+      return Flowable.fromIterable(cacheManager.withSubject(request.getSubject()).getCacheNames())
```

Comment:
Can you keep all this security stuff outside the rest resources and put them inside the  invocationHelper?  Thus by doing a ```invocationHelper.getRestCacheManager(RestRequest request)``` you'd already get a cacheManager/cache/whatever with proper security credentials